### ISSUE TITLE
Incorrect syntax linting issues reported for Adobe Coldfusion 2021 with final and static variables in components

### ISSUE THREAD

**Issue by glenfordham** [cfmleditor/CFLint#50]
The following code where a final variable is declared within a component is valid Adobe Coldfusion 2021 code, but throws an unexpected error in the linter.

<img width="360" height="147" alt="Image" src="https://github.com/user-attachments/assets/81ed0ebb-2f6a-4a39-96fe-ec0a39783b63" />
<img width="480" height="29" alt="Image" src="https://github.com/user-attachments/assets/47fd6a05-3dec-4ba4-8259-3e3cd2397268" />

Declaring static variables also throws a similar error.

<img width="360" height="95" alt="Image" src="https://github.com/user-attachments/assets/17cb6321-fae9-4865-9191-32d985a8055e" />
<img width="360" height="20" alt="Image" src="https://github.com/user-attachments/assets/8711e8a0-2802-4d4b-b190-f58f1257c27f" />

Similarly, declaring static variables in a static block throws an error:
```
component {
    static {
        public myVar = "myValue";
    }

}
```

Referencing static variables from outside the component also throws an error.

<img width="354" height="24" alt="Image" src="https://github.com/user-attachments/assets/ae98d319-2ef8-4cd1-9cf7-b41d340696e0" />
<img width="480" height="23" alt="Image" src="https://github.com/user-attachments/assets/dd6068cb-d7cf-41f8-8d93-65a89c3d2c2d" />

**Comment by ghedwards** [comment]
Fixed. Closing with the evidence.

All four constructs from this report were cfparser faults rather than CFLint ones — each reproduced directly against `CFMLParser.parseScript`, not through the linter.

| Reported case | Was | Fixed by |
|---|---|---|
| `final myConst = "c";` | already worked | — |
| `static myStatic = "v";` | **NullPointerException** | [cfparser#65](https://github.com/cfmleditor/cfparser/pull/65) |
| `static { public myVar = "v"; }` | `no viable alternative at input 'publicmyVar='` | [cfparser#66](https://github.com/cfmleditor/cfparser/pull/66) |
| `Other::sharedValue` | parsed, but `::` was discarded | [cfparser#67](https://github.com/cfmleditor/cfparser/pull/67) |

## What was actually wrong

**`static x = 1`** threw rather than reporting an error, so a single such line aborted the whole file — through CFLint that surfaced as `PARSE_ERROR: Unable to parse` and no results at all. `STATIC` was reachable only as a function modifier, so the parser committed to a function declaration, found no `function` keyword, and dereferenced null. `final` worked because it had already been given a declaration path; `static` had not.

**The static block** parsed fine without an access modifier — only `public`, `private`, `remote` and `package` on a member were rejected. Fixing that also surfaced that the block was being discarded entirely: `static { myVar = "v"; }` decompiled to plain `myVar = 'v'`, losing the static-ness.

**`::`** parsed but decompiled to `.`, turning a static reference into an ordinary property access. Checking it turned up the same fault in `?.` — `a?.b` decompiled to `a.b`, and those differ at runtime, since one yields null where the other throws. An existing fixture named `safenav` had the broken output recorded as its expected result.

## Verification

Your constructs in one component, scanned with CFLint built against the fixed parser:

```cfml
component {
    final myConst = "c";
    static myStatic = "v";
    static {
        public myPublic = "p";
        private myPrivate = "q";
    }
    function doThing() {
        undeclared = now();
        var x = Other::sharedValue;
        return myStatic & x;
    }
}
```

```
parse errors: none
Variable undeclared is not declared with a var statement.
```

The second line matters as much as the first: a parse failure used to silence every rule in the file, so "no errors" on its own would look identical to the bug. Ordinary findings still come through.

CFLint's 675 tests pass against the rebuilt parser.

## Getting the fix

CFLint pins `cfparser 2.16.0-SNAPSHOT`, which is the coordinate the fixes were published to — so this arrives by rebuild, with no version change on either side.

One caveat if you are testing this soon: Gradle treats `-SNAPSHOT` as a changing module and caches it for 24 hours. Because the version string did not change, a build shortly after the republish can still compile against the previous jar. A clean Maven resolve, or waiting out the window, gives a trustworthy answer.

## One thing not covered

`static final MAX = 10;` — both modifiers together — still does not parse. That combination does not appear in this report; I hit it while writing test code and could find no evidence either way that it is valid CFML, so I have deliberately not changed the grammar for it. If it is valid in ACF 2021, it is worth its own issue and is a small fix.

---
_Generated by [Claude Code](https://claude.ai/code)_
